File: backend/ModuMessenger/src/main/java/com/example/modumessenger/member/entity/Member.java
package com.example.modumessenger.member.entity;

import com.example.modumessenger.chat.entity.ChatRoomMember;
import com.example.modumessenger.common.domain.BaseTimeEntity;
import com.example.modumessenger.member.dto.MemberDto;
import com.example.modumessenger.member.entity.profile.Profile;
import lombok.*;

import javax.persistence.*;
import javax.validation.constraints.NotNull;
import java.util.ArrayList;
import java.util.List;
import java.util.stream.Collectors;

@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class Member extends BaseTimeEntity {

    @Id
    @Column(name = "member_id")
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotNull
    private String userId;

    private String auth;

    @Enumerated(EnumType.STRING)
    private Role role;

    @NotNull
    private String email;

    private String username;

    private String statusMessage;
    private String profileImage;
    private String wallpaperImage;

    @OneToMany(mappedBy = "member", cascade = CascadeType.ALL)
    private List<Profile> profileList = new ArrayList<>();

    @OneToMany(mappedBy = "member", cascade = CascadeType.ALL)
    private List<ChatRoomMember> chatRoomMemberList = new ArrayList<>();

    @ElementCollection
    @CollectionTable(name = "friends", joinColumns = @JoinColumn(name = "id"))
    @Column(name = "friends")
    private List<Long> Friends;

    @Override
    public String toString() {
        return getUserId() + ", " + getUsername() + "," + getEmail() + "," + getAuth() + "," + getStatusMessage() + "," + getProfileImage();
    }

    public Member update(String name, String picture) {
        this.username = name;
        this.profileImage = picture;

        return this;
    }

    public Member(MemberDto memberDto) {
        setUserId(memberDto.getUserId());
        setAuth(memberDto.getAuth());
        setRole(memberDto.getRole());
        setEmail(memberDto.getEmail());
        setUsername(memberDto.getUsername());
        setStatusMessage(memberDto.getStatusMessage());
        setProfileImage(memberDto.getProfileImage());
        setWallpaperImage(memberDto.getWallpaperImage());
        setProfileList(memberDto.getProfileDtoList().stream().map(Profile::new).collect(Collectors.toList()));
        this.Friends = new ArrayList<>();
    }

    public Member(String userId) {
        this.userId = userId;
    }

    public Member(String userId, String email) {
        this.userId = userId;
        this.email = email;
    }

    public Member(String email, String name, String picture) {
        setEmail(email);
        setUsername(name);
        setProfileImage(picture);
    }
}

File: android/modu-chat/app/src/main/kotlin/com/example/modumessenger/core/model/Member.kt
package com.example.modumessenger.core.model

/** 로그인한 사람과 친구를 같은 모양으로 다룬다. 서버의 `ResponseMemberDto`/`ResponseFriendDto` 를 합친 모습. */
data class Member(
    val id: Long = 0L,
    val userId: String = "",
    val email: String = "",
    val username: String = "",
    val statusMessage: String = "",
    val profileImage: String = "",
    val wallpaperImage: String = "",
    val role: Role = Role.USER,
    val profiles: List<Profile> = emptyList(),
    /** 친구 목록 응답에만 들어 있는 "내가 이 친구에게 붙인 이름". */
    val friendName: String? = null,
)

/** 서버 JSON 은 `"ROLE_ADMIN"`/`"ROLE_USER"` 이고, 매핑은 [com.example.modumessenger.data.dto.roleOf] 가 한다. */
enum class Role { ADMIN, USER }
